src/Parser.php

```diff
@@ -292,6 +292,10 @@ public function getParent(): ?Node
 
     public function setParent(?Node $parent): void
     {
+        if (null !== $this->parent) {
```

Comment:
```suggestion
        if (null !== $parent && null !== $this->parent) {
```

Or maybe remove the nullable here ?